src/wizards/generateProject/generationWizard.ts

```diff
@@ -58,14 +58,68 @@ export async function generateProjectWizard() {
     });
 
     state.buildTool = (await input.showQuickPick({
-      title: 'Quarkus Tools',
+      title: INPUT_TITLE,
       step: input.getStepNumber(),
       totalSteps: state.totalSteps,
       placeholder: 'Pick build tool',
       items: quickPickItems,
       activeItem: quickPickItems[0]
     })).label;
 
+    return (input: MultiStepInput) => inputPlatformVersion(input, state);
+  }
+
+  async function inputPlatformVersion(input: MultiStepInput, state: Partial<ProjectGenState>) {
+
+    const quickPickItems: PlatformVersionPickItem[] = await getCodeQuarkusApiPlatforms();
+
+    // Sort by recommended and quarkusCoreVersion in case of tie in the quick pick list
+    quickPickItems.sort((x: PlatformVersionPickItem, y: PlatformVersionPickItem) => {
+      return (x.recommended === y.recommended) ? (quarkusCoreVersionComparator(x.quarkusCoreVersion.split("."), y.quarkusCoreVersion.split(".")) ? -1 : 1) : x.recommended ? -1 : 1;
+
+      function quarkusCoreVersionComparator(x: string[], y: string[]) {
```

Comment:
This should cover any length of quarkusCoreVersion comparison

Reply:
Number.isNaN works on my machine, but has an error thrown on CI

Reply:
I would avoid writing such things from scratch. There is almost always a library that can do it. For example https://www.npmjs.com/package/semver . I think to make it simple, you could take the item's key, and use the string to the right of ':', which is seems to be a float (eg. 2.2) as opposed to a version (eg. 2.2.3).

I think parseFloat should be able to convert it.

Reply:
This would no longer sort by quarkusCoreVersion, since the quarkusCoreVersion typically comes as x.x.x, correct?